tests/components/twinkly/test_config_flow.py

```diff
@@ -1,196 +1,182 @@
 """Tests for the config_flow of the twinly component."""
 
-from unittest.mock import patch
+from unittest.mock import AsyncMock
 
-from homeassistant import config_entries
 from homeassistant.components import dhcp
-from homeassistant.components.twinkly.const import DOMAIN as TWINKLY_DOMAIN
-from homeassistant.config_entries import SOURCE_USER
+from homeassistant.components.twinkly.const import DOMAIN
+from homeassistant.config_entries import SOURCE_DHCP, SOURCE_USER
 from homeassistant.const import CONF_HOST, CONF_ID, CONF_MODEL, CONF_NAME
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResultType
 
-from . import TEST_MODEL, TEST_NAME, ClientMock
+from .const import TEST_MAC, TEST_MODEL, TEST_NAME
 
 from tests.common import MockConfigEntry
 
 
-async def test_invalid_host(hass: HomeAssistant) -> None:
-    """Test the failure when invalid host provided."""
-    client = ClientMock()
-    client.is_offline = True
-    with patch(
-        "homeassistant.components.twinkly.config_flow.Twinkly", return_value=client
-    ):
-        result = await hass.config_entries.flow.async_init(
-            TWINKLY_DOMAIN, context={"source": config_entries.SOURCE_USER}
-        )
-        assert result["type"] is FlowResultType.FORM
-        assert result["step_id"] == "user"
-        assert result["errors"] == {}
-        result = await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            {CONF_HOST: "dummy"},
-        )
-
+async def test_full_flow(
+    hass: HomeAssistant,
+    mock_twinkly_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
```

Comment:
```suggestion
@pytest.mark.usefixtures("mock_twinkly_client", "mock_setup_entry")
async def test_full_flow(hass: HomeAssistant) -> None:
```
opinionated, probably.